test/integration/evictions/evictions_test.go

```diff
@@ -184,30 +188,31 @@ func TestConcurrentEvictionRequests(t *testing.T) {
 				}
 			}
 			if e != nil {
-				if errors.IsConflict(err) {
-					fmt.Errorf("Unexpected Conflict (409) error caused by failing to handle concurrent PDB updates: %v", e)
-				} else {
-					errCh <- e
+				if errors.IsConflict(e) {
+					e = fmt.Errorf("Unexpected Conflict (409) error caused by failing to handle concurrent PDB updates: %v", e)
 				}
+				errCh <- e
 				return
 			}
 			doneCh <- true
 		}(i, doneCh, errCh)
 	}
 
 	doneCount := 0
+loop:
 	for {
 		select {
 		case err := <-errCh:
 			t.Errorf("%v", err)
-			return
+			break loop
 		case <-doneCh:
 			doneCount++
 			if doneCount == 10 {
```

Comment:
you may want to test that `doneCount` matches a previously set constant to make the test slightly less brittle.